### ISSUE TITLE
Can't access scripts in v0.6.0-dev2

### ISSUE THREAD

**Issue by twalts** [antonpup/Aurora#690]
### What are you experiencing an issue with?

- [ ] Aurora
- [ ] Aurora supported game, (Fill in the game)
- [ ] Aurora Wrapper
- [ ] Aurora Profile
- [ ] Aurora supported device, (Fill in problematic devices)
- [x] Aurora Scripts
- [ ] Aurora Updater
- [ ] Aurora Lighting issue
- [ ] Other, (Fill in another issue here)

### What kind of an issue is it?

- [ ] Crash
- [ ] Performance problems
- [ ] Visual bug
- [x] Localization issue
- [ ] Enhancement suggestion
- [ ] Device issue
- [ ] Other, (Fill in another type of issue here)

### Expected behavior:
I put a script in the scripts folder and it shows up in the dropdown menu when I add a script layer.

### Actual behavior:
I added the script to both /Roaming/Aurora/Profiles/desktop/Scripts/ and C:/Program Files/Aurora/Scripts/, but in both cases the dropdown menu remained greyed out.

### How often does this problem occur?

This problem occurs... everytime.

### Please describe the steps to reproduce this issue:
The issue can be reproduced by doing following steps:

1. 
2. 
3. 

### What is the version of Aurora you are experiencing this issue with?

My version is... 0.6.0-dev2

### Does the previous version of Aurora have this issue?
The previous version 0.6.0 did not have this issue.

### Please include any logs, screenshots, or files that are related to this issue:

For logs, you can use: http://pastebin.com/
For screenshots, you can use: http://imgur.com/
For files, you can use: http://www.tinyupload.com/ , https://www.dropbox.com/ , or attach them to the issue.

**Comment by simon-wh** [comment]
You should check the logs as there is a decent chance that the reason it isn't being displayed because it has errored out when it was being loaded.

**Comment by twalts** [comment]
@simon-wh Thanks for the quick response! You were right. Not sure exactly what the logs mean, but I found an error loading the scripts. 
```
7/26/2017 1:29:12 PM: An error occured while trying to load script C:\Program Files\Aurora\Scripts\Devices\example_py_script.py. Exception: System.Exception: Provided script, does not meet all the requirements
   at Aurora.Devices.ScriptedDevice.ScriptedDevice..ctor(Object script) in B:\Repos\Aurora\Project-Aurora\Project-Aurora\Devices\ScriptedDevice\ScriptedDevice.cs:line 36
   at CallSite.Target(Closure , CallSite , Type , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at Aurora.Devices.DeviceManager..ctor() in B:\Repos\Aurora\Project-Aurora\Project-Aurora\Devices\DeviceManager.cs:line 63

7/26/2017 1:29:13 PM: An error occured while trying to load script C:\Program Files\Aurora\Scripts\Devices\example_script.cs. Exception: System.Exception: Provided script, does not meet all the requirements
   at Aurora.Devices.ScriptedDevice.ScriptedDevice..ctor(Object script) in B:\Repos\Aurora\Project-Aurora\Project-Aurora\Devices\ScriptedDevice\ScriptedDevice.cs:line 36
   at CallSite.Target(Closure , CallSite , Type , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at Aurora.Devices.DeviceManager..ctor() in B:\Repos\Aurora\Project-Aurora\Project-Aurora\Devices\DeviceManager.cs:line 77

7/26/2017 1:29:13 PM: An error occured while trying to load script C:\Program Files\Aurora\Scripts\Devices\PingPulser.cs. Exception: Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: 'Aurora.Scripts.VoronScripts.PingPulser' does not contain a definition for 'devicename'
   at CallSite.Target(Closure , CallSite , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
   at Aurora.Devices.ScriptedDevice.ScriptedDevice..ctor(Object script) in B:\Repos\Aurora\Project-Aurora\Project-Aurora\Devices\ScriptedDevice\ScriptedDevice.cs:line 21
   at CallSite.Target(Closure , CallSite , Type , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at Aurora.Devices.DeviceManager..ctor() in B:\Repos\Aurora\Project-Aurora\Project-Aurora\Devices\DeviceManager.cs:line 77
```

**Comment by simon-wh** [comment]
@twalts The first two are outdated device scripts. But the third is what I assume is one of the scripts you've put in the device scripts. They should be put in %appdata%/Aurora/Profiles/%application%/Scripts

**Comment by twalts** [comment]
@simon-wh I put PingPulser.cs in the /Profiles/desktop/Scripts/, but the dropdown menu remains greyed out. There's no mention in the logs of an error loading it either. It worked perfectly for me last version of Aurora, but since I've upgraded it can't seem to find the .cs file.

**Comment by simon-wh** [comment]
I just tried to load some scripts and it isn't working for me either. Must've accidentally broken it at some point. I'm looking into it

**Comment by twalts** [comment]
@simon-wh Awesome, thank you!!

**Comment by simon-wh** [comment]
@twalts I've found out why it isn't working. It was a while back when I made the changes to the scripts so that's why I forgot about this. The scripts are now required to implement the IEffectScript interface and there are some slight variations with what is required by the interface and what used to be required. So any previous scripts will need to be updated to work with the dev version and in future the public 0.6.1 release. However, I will add a log output which will notify you that no compatible effects were found in the script. Because in this case it was loading the script fine but it didn't find any classes that inherited IEffectScript.
